src/inventory_client/inventory_client.go

```diff
@@ -41,20 +46,64 @@ type EnabledHostData struct {
 	Host      *models.Host
 }
 
-func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret string, logger *logrus.Logger) (*inventoryClient, error) {
+func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret string, insecure bool, caPath string, logger *logrus.Logger) (*inventoryClient, error) {
 	clientConfig := client.Config{}
 	var err error
 	clientConfig.URL, err = url.ParseRequestURI(createUrl(inventoryURL))
 	if err != nil {
 		return nil, err
 	}
 
-	clientConfig.Transport = requestid.Transport(http.DefaultTransport)
+	var certs *x509.CertPool
+	if insecure {
+		logger.Warn("Certificate verification is turned off. This is not recommended in production environments")
+	} else {
+		certs, err = readCACertificate(caPath)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	clientConfig.Transport = requestid.Transport(&http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: insecure,
+			RootCAs:            certs,
+		},
+	})
 	clientConfig.AuthInfo = auth.AgentAuthHeaderWriter(pullSecret)
 	assistedInstallClient := client.New(clientConfig)
 	return &inventoryClient{logger, assistedInstallClient, strfmt.UUID(clusterId)}, nil
 }
 
+func readCACertificate(capath string) (*x509.CertPool, error) {
+
+	if capath == "" {
+		return nil, nil
+	}
+
+	caData, err := ioutil.ReadFile(capath)
+	if err != nil {
+		return nil, err
+	}
+
+	pool := x509.NewCertPool()
+	if !pool.AppendCertsFromPEM(caData) {
+		return nil, fmt.Errorf("failed to load certificate: %s", capath)
+	}
+
```

Comment:
please log that u read it successfully

Reply:
Hmm, it's a normal flow. Do you think logging it with `debug` could be useful? Or with `info`?

Reply:
it happens once. so info is good